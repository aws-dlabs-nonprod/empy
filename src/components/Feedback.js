import React, { Component } from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import * as ActionTypes from '@constants/ActionTypes';

class Feedback extends Component {

    constructor() {
        super();
    }

    render() {
        return (
            <StyledFeedback {...this.props}>
                <div className="feedback"></div>
            </StyledFeedback>
        );
    }
}

const StyledFeedback = styled.div`
    bottom:0;
    left: 0;
    position: absolute;
    right: 0;
    top: 0;

    pointer-events: ${props => props.isVisible ? 'auto' : 'none'};
    opacity: ${props => props.isVisible ? 1 : 0};
    transition: opacity 0.3s ease-in-out;

    .feedback {
        bottom:0;
        left: 0;
        position: fixed;
        right: 0;
        top: 0;
    }
`;

function mapStateToProps(state) {
    return {
        isFinished: state.isFinished,
        isConnected: state.isConnected
    };
}

function mapDispatchToProps(dispatch) {
    return {
        feedbackSubmit: () => dispatch({
            type: ActionTypes.FEEDBACK_SUBMIT
        }),
    };
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Feedback);
