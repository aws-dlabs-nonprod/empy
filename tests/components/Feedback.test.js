import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureStore([]);

import Feedback from '@components/Feedback';

describe('Feedback', () => {

    test('renders correctly', () => {
        const store = mockStore({
            isConnected: false
        });

        const component = shallow(
            <Provider store={store}>
                <Feedback />
            </Provider>
        );

        expect(component).toMatchSnapshot();
    });
});
