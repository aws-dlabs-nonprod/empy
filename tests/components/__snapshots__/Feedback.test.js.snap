// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback renders correctly 1`] = `<Connect(Feedback) />`;
